libs/cardano-ledger-core/src/Cardano/Ledger/Core/TxCert.hs

```diff
@@ -152,6 +156,19 @@ instance NoThunks (PoolCert c)
 instance NFData (PoolCert c) where
   rnf = rwhnf
 
+instance Crypto c => ToJSON (PoolCert c) where
+  toJSON = \case
+    RegPool poolParams ->
+      object
+        [ "kind" .= String "RegPool"
+        , "poolParams" .= toJSON poolParams
+        ]
+    RetirePool poolId epochNo ->
+      object
+        [ "kind" .= toJSON poolId
+        , "poolParams" .= toJSON epochNo
```

Comment:
Is it "epochNo" here instead of "poolParams"  ?

Reply:
Yes, also  `"kind"` should be set to `"RetirePool"`, not `poolId`.

Done